### PR TITLE
[Parse] Accept 'self' after 'each'

### PR DESCRIPTION
**Pull request by rintaro**
Also move 'repeat', 'each', and 'any' expression parsing to 'parseExprSequenceElement'

rdar://107450487

### CODE REVIEW

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/1895
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please Test Source Compatibility

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/1895
@swift-ci Please Test Source Compatibility

File: lib/Parse/ParseExpr.cpp [hunk 1]
```diff
@@ -465,6 +465,44 @@ ParserResult<Expr> Parser::parseExprSequenceElement(Diag<> message,
...
+    auto patternExpr = parseExprImpl(
+        diag::expected_expr_after_repeat, isExprBasic);
+    if (patternExpr.isNull())
+      return patternExpr;
+
+    auto *expansion =
+        PackExpansionExpr::create(Context, repeatLoc, patternExpr.get(),
+                                  /*genericEnv*/ nullptr);
+    return makeParserResult(expansion);
+  }
```
Comment (ahoppen) [inline lib/Parse/ParseExpr.cpp:489]: The same comment that I left on https://github.com/apple/swift-syntax/pull/1895/files#r1259240007

> I just noticed that we already never parse `repeat` as a function call in the following (not a regression introduced by this PR, this just made me realize it…
> 
> ```swift
> func repeat() {}
> repeat()
> ```

**Review by ahoppen — APPROVED** [review]

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/1895
@swift-ci Please smoke test

**Pull request merged**